src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -123,6 +125,31 @@ public string HostId
             }
         }
 
+        /// <summary>
+        /// Gets the host machine ID. All host instances on the same machine will have the same HostMachineId.
+        /// </summary>
+        public string HostMachineId
+        {
+            get
+            {
+                if (_hostMachineId == null)
+                {
+                    _hostMachineId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X", CultureInfo.InvariantCulture).PadLeft(32, '0');
+
+                    _hostMachineId = _hostMachineId.Substring(0, 32).ToLowerInvariant();
+                }
+
+                return _hostMachineId;
+            }
+
+            // for testing
+            internal set
```

Comment:
Beware, each storage entity has different naming rules.  A summary here: [https://blogs.msdn.microsoft.com/jmstall/2014/06/12/azure-storage-naming-rules/](https://blogs.msdn.microsoft.com/jmstall/2014/06/12/azure-storage-naming-rules/)

---
In reply to: [92639500](https://github.com/Azure/azure-webjobs-sdk/pull/913#discussion_r92639500) [](ancestors = 92639500)